### PR TITLE
merge request and response actions; support multiple actions

### PR DESCRIPTION
**Pull request by Hexilee**
Signed-off-by: hexilee <i@hexilee.me>

Currently, the encoded action is not suitable to interact with chaos-daemon. The json-encoded `Action::Abort` is `"abort"` while `Action::Delay(1s)` will be encoded into `{ "delay": "1s" }`.

So this PR merges all actions into one struct.

### CODE REVIEW

**Comment by Andrewmatilde** [comment]
While `RawConfig` is more and more close to `tproxy::config` , i think it is possible to be a good idea to merge it and use default macro of serde.It will reduce redundancy caused by same part of there two struct.

File: src/cmd/config.rs [hunk 1]
```diff
@@ -272,8 +204,26 @@ impl TryFrom<RawResponseSelector> for ResponseSelector {
...
+        })
+    }
+}
+
+impl TryFrom<RawReplaceAction> for ReplaceAction {
+    type Error = Error;
+
+    fn try_from(raw: RawReplaceAction) -> Result<Self, Self::Error> {
+        Ok(Self {
+            path: raw.path.as_ref().map(|path| path.parse()).transpose()?,
```
[the remark this answers is not in the document]
Comment (Hexilee) [reply src/cmd/config.rs:216]: No, it's unnecessary.

[the remark this answers is not in the document]
Comment (Hexilee) [reply src/cmd/config.rs]: Removed

File: src/handler.rs [hunk 2]
```diff
@@ -10,73 +10,53 @@ use tokio::time::sleep;
...
-    pub action: ResponseAction,
-}
-
-#[derive(Debug, Eq, PartialEq, Clone)]
-pub struct RequestSelector {
+pub struct Selector {
     pub port: Option<u16>,
     pub path: Option<PathAndQuery>,
     pub method: Option<Method>,
     pub headers: Option<HeaderMap>,
```
[the remark this answers is not in the document]
Comment (Hexilee) [reply src/handler.rs]: Renamed

**Comment by Hexilee** [comment]
> While `RawConfig` is more and more close to `tproxy::config` , i think it is possible to be a good idea to merge it and use default macro of serde.It will reduce a very small redundancy caused by same part of there two struct.

Maybe it needs more workload to use a custom field converter in serde.

**Comment by Andrewmatilde** [comment]
LGTM

**Pull request merged**